### PR TITLE
feat: context injection into chat, MCP tools, and discovery (IND-312)

### PR DESCRIPTION
**Pull request by yanekyuk**
## Summary
- **Extended `getNetwork`** interface and adapter to return `prompt`, `type`, `metadata`, and `permissions` — the protocol layer now has full network data when resolving chat context
- **Replaced JSON serialization** of scoped index in chat system prompt with `renderNetworkContext()` markdown, including membership role — LLM now receives structured event metadata (dates, themes, schedule) instead of raw JSON
- **Added `renderedContext`** field to each network entry in the `read_networks` MCP tool response — MCP clients get pre-rendered markdown alongside raw data
- **Injected network context** into the opportunity evaluator's input, gated by `permissions.contextInjection.discovery` (default: true) — event network metadata enriches opportunity scoring

Closes IND-312

## Test plan
- [x] Protocol `tsc --noEmit` passes
- [x] Backend `tsc --noEmit` passes
- [x] Protocol build succeeds
- [x] Existing renderer tests pass (7/7)
- [ ] Copilot review

### CODE REVIEW

File: packages/protocol/src/shared/interfaces/database.interface.ts [hunk 1]
```diff
@@ -697,7 +697,14 @@ export interface Database {
   /**
    * Get index by ID (id and title only). Used for opportunity presentation.
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/shared/interfaces/database.interface.ts]: Fixed. Updated doc comments on both interface declarations to say "core fields" instead of "id and title only".

File: packages/protocol/src/network/network.tools.ts [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+  const enrichWithContext = (networks: Array<Record<string, unknown>>) =>
+    networks.map((n) => ({
+      ...n,
+      renderedContext: renderNetworkContext({
+        type: (n.type as string) ?? 'community',
+        title: (n.title as string) ?? '',
+        prompt: n.prompt as string | undefined,
+        metadata: (n.metadata as Record<string, unknown>) ?? {},
+      }),
+    }));
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/network/network.tools.ts:21]: The renderer itself has 7 unit tests covering all network types and edge cases. `enrichWithContext` is a 6-line passthrough that calls the renderer — adding a test for it verifies the call-through, not new logic. Skipping.

File: packages/protocol/src/opportunity/opportunity.evaluator.ts [hunk 3]
```diff
@@ -473,7 +475,10 @@ CRITICAL SCORING RULES FOR DISCOVERY REQUESTS:
   RAG SCORE: ${e.ragScore ?? '—'}
   MATCHED VIA: ${e.matchedVia ?? '—'}`;
     }).join('\n');
-    const humanContent = `DISCOVERER: ${input.discovererId}${introModePart}${discoveryQueryPart}\n\nENTITIES:\n${entitiesBlock}${existingPart}`;
+    const networkContextPart = input.networkContexts && Object.keys(input.networkContexts).length > 0
+      ? `\n\nNETWORK CONTEXTS:\n${Object.values(input.networkContexts).join('\n\n')}`
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.evaluator.ts]: Fixed. Switched to `Object.entries` — each context block is now prefixed with `[INDEX: <uuid>]` so the LLM can associate it with entity rows.

File: packages/protocol/src/opportunity/opportunity.graph.ts [hunk 4]
```diff
@@ -143,6 +144,33 @@ export function buildDiscovererContext(
...
+): Promise<Record<string, string>> {
+  const networkIds = [...new Set(entities.map((e) => e.networkId))];
+  const contexts: Record<string, string> = {};
+  for (const nid of networkIds) {
+    const network = await database.getNetwork(nid);
+    if (!network) continue;
+    const perms = (network.permissions ?? {}) as Record<string, unknown>;
+    const injection = perms.contextInjection as { discovery?: boolean } | undefined;
+    if (injection?.discovery === false) continue;
+    contexts[nid] = renderNetworkContext({
```
[the remark this answers is not in the document]
Comment (yanekyuk) [reply packages/protocol/src/opportunity/opportunity.graph.ts:164]: Fixed. Replaced sequential `for` loop with `Promise.all` over the unique network IDs.
